starlette/testclient.py

```diff
@@ -263,7 +263,9 @@ def __init__(self, app: ASGI3App, scope: Scope) -> None:
         self.app = app
         self.scope = scope
         self.accepted_subprotocol = None
-        self._loop = asyncio.new_event_loop()
+        self._loop = asyncio.get_event_loop()
+        if self._loop.is_running():
+            self._loop = asyncio.new_event_loop()
```

Comment:
I think this use the same method as the TestClient (https://github.com/blueyed/starlette/blob/8fd8152b1536dc42d1d122791f0ee4cdeec0aaa1/starlette/testclient.py#L232-L236) - and therefore probably get factored out into a function/method of its own.

Reply:
I was initially going to do so, but they have some differences.
In `TestClient` exception is raised on calling `get_event_loop` because there is no current event loop at all. 
In `WebSocketTestSession` no exception is raised on calling `get_event_loop`. However if current event loop is running there will be RuntimeError in [_run](https://github.com/unmade/starlette/blob/487-testclient-websocket-same-loop/starlette/testclient.py#L289) method. This `_run` method runs in thread, so I can't do fallback here. 

Condition `if self._loop.is_running` is tested by [here](https://github.com/unmade/starlette/blob/487-testclient-websocket-same-loop/tests/test_testclient.py#L132-L134) where you can see nested calls to endpoints each of which creates new `WebSocketTestSession`

Reply:
Ah, it is the other way around..
(reminded me of https://github.com/encode/starlette/pull/505 btw - maybe you can provide some insight there / help with it)

Reply:
Sure, I replied you in #505

Reply:
Please merge it!!!😭,otherwise I can only edit files by myself,it feels insecure😂. In my code,I  edit the _run method as this:
``` python
if loop.is_running():
    loop.call_soon(self.app, scope, receive, send)
else:
    loop.run_until_complete(self.app(scope, receive, send))
```
It works,maybe there is more appropriate way to deal with this.